### DISCUSSION TITLE
DataGridTemplateColumn.Width binding error with DataGridLength property

### DISCUSSION THREAD

**Discussion by MichaelSvejcar** [AvaloniaUI/Avalonia#11107] [Q&A]
I am encountering a binding error when I try to bind the Width property of a DataGridTemplateColumn to a DataGridLength property in my ViewModel. The error message I receive is:

> Severity    Code    Description    Project    File    Line    Suppression State
> Error       Unable to find suitable setter or adder for property Width of type Avalonia.Controls.DataGrid:Avalonia.Controls.DataGridColumn for argument Avalonia.Markup:Avalonia.Data.Binding, available setter parameter lists are:
> Avalonia.Controls.DataGridLength Line 167, position 69.    DropNormalizerGUI_Avalonia    C:\CSharp\DropNormalizer\DropNormalizerGUI_Avalonia\Views\MainWindow.axaml    167

The issue appears on this line of XAML code:
`<DataGridTemplateColumn Header="{Binding WaveImage.Header}" Width="{Binding WaveImage.Width}"  IsVisible="{Binding WaveImage.IsVisible, Mode=TwoWay}">`

My ViewModel has a ColumnInfo class with a DataGridLength property for Width:
public class ColumnInfo : INotifyPropertyChanged
```
public class ColumnInfo : INotifyPropertyChanged
{
    // ... Other properties and methods ...

    public DataGridLength Width
    {
        get
        {
            return width;
        }
        set
        {
            width = value;
            OnPropertyChanged(this, nameof(Width));
        }
    }
    private DataGridLength width;
}
```

The other bindings, like WaveImage.Header and WaveImage.IsVisible, work fine. I am confused as to why the binding for the Width property is not working, since both the DataGridTemplateColumn.Width property and the ColumnInfo.Width property are of type DataGridLength. Can anyone help me understand and resolve this issue?

**Comment by timunie** [comment]
It is not an `AvaloniaProperty` and therefore not bindable, see: https://github.com/AvaloniaUI/Avalonia/blob/813de5745f92668360f67957a8211d97ff18d16a/src/Avalonia.Controls.DataGrid/DataGridColumn.cs#L562-L605 

You can open a feature request if there is none already, but it will take some time to implement and most likely will not happen for 11.0. 

For a faster solution I'd suggest to create a custom Behavior which you can re-use. See https://github.com/AvaloniaUI/Avalonia.Xaml.Behaviors
